pkg/archive/fuzz_test.go

```diff
@@ -0,0 +1,44 @@
+package archive
+
+import (
+	"bytes"
+	"os"
+	"testing"
+
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
+)
+
+func FuzzDecompressStream(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		r := bytes.NewReader(data)
+		_, _ = DecompressStream(r)
+	})
+}
+
+func FuzzUntar(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		ff := fuzz.NewConsumer(data)
+		tarBytes, err := ff.TarBytes()
+		if err != nil {
+			return
+		}
+		options := &TarOptions{}
+		err = ff.GenerateStruct(options)
+		if err != nil {
+			return
+		}
+		tmpDir := t.TempDir()
+		Untar(bytes.NewReader(tarBytes), tmpDir, options)
+	})
+}
+
+func FuzzApplyLayer(f *testing.F) {
+	f.Fuzz(func(t *testing.T, data []byte) {
+		defer os.Remove("testDir")
+		err := os.Mkdir("testDir", 0750)
+		if err != nil {
+			return
+		}
+		_, _ = ApplyLayer("testDir", bytes.NewReader(data))
```

Comment:
Looks like I missed this one 😅 

```suggestion
		tmpDir := t.TempDir()
		_, _ = ApplyLayer(tmpDir, bytes.NewReader(data))
```